flash/vision/segmentation/transforms.py

```diff
@@ -0,0 +1,70 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Callable, Dict, List, Tuple
+
+import kornia as K
+import torch
+import torch.nn as nn
+
+
+class ApplyTransformToKeys(nn.Sequential):
+
+    def __init__(self, keys: List[str], *args):
+        super().__init__(*args)
+        self.keys = keys
+
+    @torch.no_grad()
+    def forward(self, x: Dict[str, torch.Tensor]) -> Dict[str, torch.Tensor]:
+        for seq in self.children():
+            for aug in seq:
+                for key in self.keys:
+                    # check wether the transform was applied
+                    # and apply transform
+                    if hasattr(aug, "_params") and bool(aug._params):
```

Comment:
Can you explain what this is doing?

Reply:
in kornia every time a forward pass is called for an augmentation - the generated random params are stored in `_params` in order to assure deterministic transformations. We reuse those parameters to be applied across the different key values.

Reply:
I see, makes sense :smiley: